civis/io/_tables.py

```diff
@@ -774,6 +796,23 @@ def csv_to_civis(filename, database, table, api_key=None, client=None,
         Whether or not the first row of the file should be treated as
         headers. The default, ``None``, attempts to autodetect whether
         or not the first row contains headers.
+    primary_keys: list[str], optional
+        A list of the primary key column(s) of the destination table. If
+        existing_table_rows is "upsert", this field is required.
+    last_modified_keys: list[str], optional
+        A list of the columns indicating a record has been updated. If
+        existing_table_rows is "upsert", this field is required.
+    escaped: bool, optional
+        A boolean value indicating whether or not the source file has quotes
+        escaped with a backslash. Defaults to false.
+    execution: string, optional, default "immediate"
+        One of "delayed" or "immediate". If "immediate", refresh column
+        statistics as part of the run. If "delayed", flag the table for a
+        deferred statistics update; column statistics may not be available
+        for up to 24 hours. In addition, if existing_table_rows is "upsert",
+        delayed executions move data from staging table to final table after a
+        brief delay, in order to accommodate multiple concurrent imports to the
+        same destination table.
```

Comment:
Just to check my understanding:

Should `execution` always be `delayed` if passing multiple file ids when `existing_table_rows = "upsert"`? 
Could there ever be a situation when the delay is not long enough when upserting? What happens from the user perspective in that case?

Reply:
cleared this up offline

Reply:
Talked offline, but leaving comments for posterity/general understanding:

1. `Execution` can be both `delayed` and `immediate` with multiple files - the parameter is more to control the interplay of multiple jobs writing to the same table concurrently.
2. In `delayed` mode, each subsequent attempt to upsert pushes back the final transfer of data - data is moved from staging to final table 15 minutes after the last requested upsert.